lib/browser/storage.js

```diff
@@ -22,17 +22,25 @@ try {
 
 export const canStoreURLs = hasStorage;
 
-export function setItem(key, value) {
-  if (!hasStorage) return;
-  return localStorage.setItem(key, value);
-}
+export class DefaultStorage {
+  setItem (key, value) {
+    return new Promise((resolve, reject) => {
```

Comment:
We are currently using callbacks for the event system. What do you think if we do not use Promises to keep this consistency?